cmd/broadcast/utils.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"bufio"
+	"encoding/hex"
+	ida "github.com/harmony-one/libunison/internal/ida/coopcast"
+	"io"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const pubKeySize int = 20
+
+// Entry is a single config of a node.
+type Entry struct {
+	Sid     string
+	IP      string
+	TCPPort string
+	UDPPort string
+	PubKey  string
+	Role    string
+}
+
+// Config is a struct containing multiple Entry of all nodes.
+type Config struct {
+	config []Entry
+}
+
+// NewConfig returns a pointer to a Config.
+func NewConfig() *Config {
+	config := Config{}
+	return &config
+}
+
+// GetPeerInfo returns the selfPeer, peerList, allPeers from config instance, which used to create node instance
+func (config *Config) GetPeerInfo() (ida.Peer, []ida.Peer, []ida.Peer) {
+	var allPeers []ida.Peer
+	var peerList []ida.Peer
+	var selfPeer ida.Peer
+	for _, entry := range config.config {
+		sid, err := strconv.Atoi(entry.Sid)
+		if err != nil {
+			log.Printf("cannot convert sid")
+		}
+		peer := ida.Peer{IP: entry.IP, TCPPort: entry.TCPPort, UDPPort: entry.UDPPort, PubKey: entry.PubKey, Sid: sid}
+		if entry.Role == "0" {
```

Comment:
These magic numbers (0/1/2) need to be explained.  Or perhaps it should be renamed into a more descriptive keyword such as `self`, `peer` (with which this node has a connection), and `other` (with which this node does not have a connection).

Reply:
role is changed to enum